packages/editor-ui/src/mixins/workflowHelpers.ts

```diff
@@ -706,6 +707,10 @@ export const workflowHelpers = defineComponent({
 
 			// Workflow exists already so update it
 			try {
+				// Prevent saving if nothing has changed and not forced
+				if (!forceSave && !hasChanges) {
```

Comment:
hmm seems like a bandaid.. is this temp fix? did we find the root cause of the issue?

Reply:
How come? Issue is that we're sending a PATCH request for a specific WF id with empty data when pressing CMD+S during Workflow load. Not sending this request when workflow has no changes(or is loading) seems like appropriate fix

Reply:
oh when the workflow is loading? should not it depend on the loading state then?

Reply:
The `stateIsDirty` is always `false` on load so that would already cover it. But checking for `stateIsDirty` would also prevent saving when there's no changes. So it follows the visual pattern where "Save" button is disabled if there's no changes. But before this fix, you'd be able to save the workflow using the keyboard shortcut, which is also a bug

Reply:
idk if it's always false.. are you sure? if I recall correctly, we set `stateIsDirty` to true every time we add a node, then reset it to false after the workflow is done loading

Reply:
Just checked via this watcher:
```
stateIsDirty: {
			immediate: true,
			handler(e) {
				console.log('State is dirty? ', e);
			},
		},
```
It's false until we add/move node/connection

https://github.com/n8n-io/n8n/assets/12657221/ad11d842-f075-485e-84be-43f2df570a96

Reply:
perfect.. thanks

Reply:
would be nice to have tests too